tests/tests.fish

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env fish
+
+set BASE_DIR (dirname (status -f))
+
+function test_program
+    set TEST_SCRIPT $argv[1]
+    set TEST_MODE $argv[2]
+    set TEST_ARG $argv[3]
+    set TEST_EXPECTED $argv[4]
+
+    set tmpfile (mktemp)
+    echo "$TEST_SCRIPT" >"$tmpfile"
+
+    # Fish shell's '$_' variable doesn't behave the same when sourcing from a command 
+    # substitution, so setting an environment variable is a hacky workaround
+    if test "$TEST_MODE" = "source"
+        set -x IS_SOURCED true
+        set TEST_OUT (fish -c "source $tmpfile $TEST_ARG" | string collect)
+        set -e IS_SOURCED
+    else if test "$TEST_MODE" = "direct"
+        chmod +x "$tmpfile"
+        set TEST_OUT (fish -c "$tmpfile $TEST_ARG" | string collect)
+    else if test "$TEST_MODE" = "envgrep"
+        set -x IS_SOURCED true
+        set TEST_OUT (fish -c "source $tmpfile $TEST_ARG; env | grep TEST_AWS_CLI_MFA" | string collect)
```

Comment:
I can get some of the tests working when I put them in the `.sh` tests script, but there are a lot of differences that would be tricky to work around.  For example, for fish instead of `&& env` it needed to be `; env`.  Separate test file is better than nothing.